### ISSUE TITLE
Custom icons are not working

### ISSUE THREAD

**Issue by keem85** [nguyenquyhy/Flight-Tracker-StreamDeck#418]
Below are some useful information that you can include to help with troubleshooting the issue:
- Stream Deck Plugin and MSFS version: 1.35.21.0 MSFS, and 6.5.0 (19904) StreamDeck+

- Description of the issue & expected behavior: On Generic Button, it does not work to have ON/OFF icon. It only chooses the OFF picture, and it does not dynamically change when the button is pressed. Icons are correct .png format and size. Tested with many, many icons and the same happens every time.

- Steps to reproduce the issue: Set two different pictures on ON and OFF, under Generic Toggle Button

- If the issue happens consistently, an export of your profile would help tremendously: Doesn't matter, since it happens on every profile.

- Log files in folder `%appdata%\Elgato\StreamDeck\Plugins\tech.flighttracker.streamdeck.sdPlugin\logs\`: There is no such file path for the flight tracker plugin
- Screenshots
![123](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/assets/17292106/f6708a26-c5af-42d5-ab85-18c5c6db29ce)

**Comment by nguyenquyhy** [comment]
May I know what aircraft and what Feedback value you are using?

**Comment by keem85** [comment]
> May I know what aircraft and what Feedback value you are using?

What is a feedback value? I have no feedback value in the box.. I'm not using any spesific aircraft per se, since this happens on any aircraft/buttons... but in this case I am setting up "ADF1_RADIO_SWAP" to work as AP heading hold for the HPG 145. I am just trying to set up the generic toggle button on my Stream Deck+, and it doesn't want to change pictures from ON-picture to OFF-picture
![fb1](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/assets/17292106/3527e74b-b7a1-449e-ace1-d214da87df59)
.

I have also read the manual, and I don't understand what many of the values does, so I have only focused on the simple toggle-value and the on/off picures (which aren't working)

**Comment by nguyenquyhy** [comment]
You need to have `Feedback value` so the plugin know what is the current state (in order to show the ON/OFF images accordingly).

I don't use the 145 so I don't really understand this statement `I am setting up "ADF1_RADIO_SWAP" to work as AP heading hold for the HPG 145`, but I think you will need to put a SimConnect or LVar on feedback value that indicate the current AP heading hold status. For example, the default SimConnect value for default aircraft is `AUTOPILOT HEADING LOCK`. You can try to put that in Feedback value, but I'm not sure if that will work for HPG aircraft as HPG has some custom systems.

**Comment by keem85** [comment]
> You need to provide `Feedback value` so the plugin know what the current state is (in order to show the ON/OFF images accordingly).
> 
> I don't use the 145 so I don't really understand this `I am setting up "ADF1_RADIO_SWAP" to work as AP heading hold for the HPG 145`, but I think you will need to put a SimConnect or L-Var into feedback value that indicates the current AP heading hold status. For example, the SimConnect value for default aircraft is `AUTOPILOT HEADING LOCK`. You can try to put that into Feedback value, but I'm not sure if that will work for HPG aircraft as HPG has some custom systems.

Ahaa, I see. So this is not actually a bug, which I originally thought, but the lack of "feedback value" ?
Do you have a video that explains to me what a "feedback value" is, or "parameters"? Because I've read the documents, and I don't fully understand what these are, and where to find them. Might be my english cutting short when it comes to simple terminology, but perhaps an eli5 video would help me understand this. I'm not sure what SimConnect or L-Var is either, so this Flight Tracker really seem to be for very advanced users?

_On a seperate note, the H146 uses a third party software, since MSFS bindings doesn't exist for this aircraft. So it has it's internal bindings, and we assign obscure MSFS bindings (that are not otherwise used) in to this program, as a sort of a "proxy" or what I should call it._

**Comment by nguyenquyhy** [comment]
> Do you have a video that explains to me what a "feedback value" is, or "parameters"?

I don't have any, but if you search on YouTube for `flight tracker stream deck plugin`, you should find a few.

> this Flight Tracker really seem to be for very advanced users?

I try to cater for both new and more adept MSFS users with preset and generic buttons. 
- Preset buttons are ones where users can simply drag, drop and choose some settings for specific aircraft functionalities.
- Generic buttons are ones where users can do a lot more by choosing what happen when a button is pressed/held and what value a button should indicate. Generic buttons do require a bit more understanding on the internal of MSFS, which I will try to explain below.

> I'm not sure what SimConnect or L-Var is either

Essentially, in order for a button to show ON/OFF states, it needs to be able to read the corresponding values from the sim.
There are 2 types of values that the plugin can handle:
- SimConnect variables: Those are the values that are exposed by default aircraft systems. You can find them here: https://docs.flightsimulator.com/html/Programming_Tools/SimVars/Simulation_Variables.htm. For example, for your case you can try to use `AUTOPILOT HEADING LOCK` to get the current state of AP Heading for many aircraft.
- L-Var: Those are internal values that each aircraft creates, so there is no master list for L-Var. You will have to find this list from the aircraft developer, or use more advanced tool like FSUIPC to extract the list.

**Comment by keem85** [comment]
> > Do you have a video that explains to me what a "feedback value" is, or "parameters"?
> 
> I don't have any, but if you search on YouTube for `flight tracker stream deck plugin`, you should find a few.
> 
> > this Flight Tracker really seem to be for very advanced users?
> 
> I try to cater for both new and more adept MSFS users with preset and generic buttons.
> 
> * Preset buttons are ones where users can simply drag, drop and choose some settings for specific aircraft functionalities.
> * Generic buttons are ones where users can do a lot more by choosing what happen when a button is pressed/held and what value a button should indicate. Generic buttons do require a bit more understanding on the internal of MSFS, which I will try to explain below.
> 
> > I'm not sure what SimConnect or L-Var is either
> 
> Essentially, in order for a button to show ON/OFF states, it needs to be able to read the corresponding values from the sim. There are 2 types of values that the plugin can handle:
> 
> * SimConnect variables: Those are the values that are exposed by default aircraft systems. You can find them here: https://docs.flightsimulator.com/html/Programming_Tools/SimVars/Simulation_Variables.htm. For example, for your case you can try to use `AUTOPILOT HEADING LOCK` to get the current state of AP Heading for many aircraft.
> * L-Var: Those are internal values that each aircraft creates, so there is no master list for L-Var. You will have to find this list from the aircraft developer, or use more advanced tool like FSUIPC to extract the list.

Thank you for taking time to write this. I will dig more into this, based on this info. Thank you. Sorry I mistakenly marked this as a bug
